File: Python/array/TwoSum.py
'''
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

You can return the answer in any order.
No question of sorting as indices is required

Input = [3,3] , target = 6
Output:- [0,1]
'''
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        #return indices or two numbers
        
        for index,value in enumerate(nums):
            if target - value in hashMap:
                return {index,hashMap[target-value]}
            hashMap[value] = index
        


 

File: Apple/C++/Two Sum.cpp
/*
Question:- Return Indices of the two number and not the number
[3,3] Target:- 6
Ans:- [0,1]
Will still work

Clarifying question :- Do we need to return the number or only the indices?

Constraints-Memorize:-
You may assume that each input would have exactly one solution, and you may not use the same element twice
Only one valid answer exists.
*/

#include<bits/stdc++.h>
using namespace std;

class Solution
{
   public:
   vector<int>twosum(vector<int>&nums,int target)
    {
      unordered_map<int,int> hash;
      int length = nums.size();
      for(int i=0;i<length;i++)
       {
         if(hash.count(target-nums[i]))
         {
             return {hash[target-nums[i]],i};
          }
        hash[nums[i]]=i;
      }
    }
 };
